andela/pulse-mobile-app
# [#156569762]Add Login backend

#### What does this PR do?
Add google login feature

#### Description of Task to be completed?
- Add logic to handle Google auth 
- Add deep links to get authorization token
- Persisting token in async storage
- write tests for the feature

#### How should this be manually tested?
- run `react-native run-android` and `react-native run-ios`
- App should launch login screen
- Pressing the Login button should open google login form in  Safari view in iOS or browser for android
- Login with `test user credentials`  safari view is closed on iOS and you are redirected to  app PartnerList screen. For android browser will redirect back to app, launch it and take you to PartnerList screen
- Login with non-Andela gmail account  safari view is closed on iOS and you are redirected to  app Login screen with toast message **`Not Authorized!`**
- Kill the app instance on both android and iOS and relaunch it. It should take you back to the previous screen you were.
- Press on Logout button to Logout

#### Any background context you want to provide?

#### What are the relevant pivotal tracker stories?
[#156569762](https://www.pivotaltracker.com/story/show/156569762)

#### Screenshots (if appropriate)
![Login iOS](https://user-images.githubusercontent.com/15029943/39075195-3bb4848c-44fd-11e8-9bbd-d9982ea1d7bb.gif)
![Login Android](https://user-images.githubusercontent.com/15029943/39075183-2b68e08c-44fd-11e8-8210-a333e003b490.gif)

#### Check List

#### Questions:
